src/main/java/com/es/phoneshop/model/attributesHolder/HttpSessionAttributesHolder.java

```diff
@@ -0,0 +1,24 @@
+package com.es.phoneshop.model.attributesHolder;
+
+import jakarta.servlet.http.HttpSession;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class HttpSessionAttributesHolder implements AttributesHolder {
+
+    private final HttpSession httpSession;
+
+    @Override
+    public Object getAttribute(String name) {
+        synchronized (httpSession) {
```

Comment:
why do we need sync on adding/getting attributes to session?

Reply:
can't user make multiple  simultaneous requests to the server by opening multiple tabs or windows in their browser (or some other way)?

Reply:
agree
usually not so much attributes are stored in session, only the most important ones (as Cart in your example, or some special tokens for protection of the vulnerabilities), and for sure while adding/retrieving those objects we should sync them by the session